build/dsc/templates/dsc_type.rb.erb

```diff
@@ -4,6 +4,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
   require Pathname.new(__FILE__).dirname + '../../' + 'puppet/type/base_dsc'
 
   provide :powershell, :parent => Puppet::Type.type(:base_dsc).provider(:powershell) do
+    confine :operatingsystem => :windows, :powershell_version => '5.0.10514.6'
```

Comment:
I don't know what the default `RefreshMode` is in the Windows 10 WMF release, as the VM i have isn't fresh, but it does prevent using `Invoke-DscResource` when `RefreshMode` is not `Disabled`

~~~
[jpogran@foo-test]$ Invoke-DscResource -Name file -Method set -Property @{ ensure = 'present'; destinationpath = 'c:\foo.txt'; contents= 'bar' } -Verbose
VERBOSE: Performing the operation "Invoke-CimMethod: Resourceset" on target "MSFT_DSCLocalConfigurationManager".
VERBOSE: Perform operation 'Invoke CimMethod' with following parameters, ''methodName' = Resourceset,'className' =
MSFT_DSCLocalConfigurationManager,'namespaceName' = root/Microsoft/Windows/DesiredStateConfiguration'.
VERBOSE: An LCM method call arrived from computer FOO with user sid S-1-5-21-1787757173-3380529767-833788528-1001.
VERBOSE: Operation 'Invoke CimMethod' complete.
Invoke-DscResource : Invoke-DscResource cmdlet is not supported in Push mode. Please set the LCM RefreshMode to DISABLED.
At line:1 char:1
+ Invoke-DscResource -Name file -Method set -Property @{ ensure = 'pres ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotImplemented: (root/Microsoft/...gurationManager:String) [Invoke-DscResource], CimException
    + FullyQualifiedErrorId : MI RESULT 7,Microsoft.PowerShell.DesiredStateConfiguration.Commands.InvokeDscResourceMethodCommand
~~